### DISCUSSION TITLE
How to create videosource

### DISCUSSION THREAD

**Discussion by padi-dev-anhntn** [shogo4405/HaishinKit.kt#313] [Q&A]
Hello,
Our problem is using a 3rd party to create a filter for the face then push it to the stream via rtmp. I have successfully used haishinkit for ios but I have not been able to use it successfully on android.
I am trying to create a videosource for Haishinkit but cannot find any examples or solutions that work.

- **Input**: We have available inputs as surface or frameAvailable

`public void frameAvailable(Image image) {
    }`
![image](https://github.com/user-attachments/assets/c05c788f-aaa8-47c2-8d81-7c3f827089f0)

- **Output** : We want a suitable videosource output for HaiShinKit

-----------------
Source code demo : We are using with the following source example  [source DeepAR](https://github.com/DeepARSDK/quickstart-android-java)
Update config in MainActivity.java with this API key 
 `        deepAR.setLicenseKey("REDACTED");
`

**Comment by padi-dev-anhntn** [comment]
@shogo4405 please help us. We appreciate your help.

**Reply by shogo4405** [reply]
What can I help you with? If you would like me to create such a feature, I charge $100 USD per hour. As for the API Key, I don't think it's good for a third party to use it, so I'll remove it.

**Reply by padi-dev-anhntn** [reply]
@shogo4405 
Oh, the api_key I gave you is free and limited. I wish you could reproduce it as soon as possible. But thanks for what you did.
We don't want to bother you or ask you to create a new feature. We hope you can guide us on how to handle this issue.

**Reply by shogo4405** [reply]
This kind of guidance is only provided to certain contributors and those with paid contracts. It might be best to give up.

**Comment by vfa-khaidt1** [comment]
Have you found a solution for this?

**Reply by padi-dev-anhntn** [reply]
@vfa-khaidt1 use library [RootEncoder](https://github.com/pedroSG94/RootEncoder)
 instead
